File: ops/backend/app/schemas/event.py
"""System event schemas."""

from typing import Literal

from pydantic import BaseModel

from app.schemas.common import Severity

EventCategory = Literal[
    "trade", "strategy", "machine", "broker", "system", "database", "risk", "data"
]


class SystemEvent(BaseModel):
    """A single line in the live event terminal."""

    id: str
    time: str
    category: EventCategory
    severity: Severity
    source: str
    message: str
    machineId: str | None = None
    eventType: str | None = None
    strategy: str | None = None
    symbol: str | None = None
    sessionId: str | None = None
    sequenceId: int | None = None
    payloadSummary: str | None = None
    sourceEventType: str | None = None

File: ops/frontend/src/types/event.ts
import type { Severity } from './common'

/** Domain a live event originates from — drives the terminal colour coding. */
export type EventCategory =
  | 'trade'
  | 'strategy'
  | 'machine'
  | 'broker'
  | 'system'
  | 'database'
  | 'risk'
  | 'data'

/** A single line in the Bloomberg-style live event terminal. */
export interface SystemEvent {
  id: string
  /** ISO timestamp. */
  time: string
  category: EventCategory
  severity: Severity
  /** Originating machine / strategy / broker label. */
  source: string
  message: string
  machineId?: string | null
  eventType?: string | null
  strategy?: string | null
  symbol?: string | null
  sessionId?: string | null
  sequenceId?: number | null
  payloadSummary?: string | null
}
